controlpanel/api/aws.py

```diff
@@ -221,32 +224,31 @@ class S3AccessPolicy:
 
     def __init__(self, policy):
         self.policy = policy
-        self._policy_document = None
         self.statements = {}
 
-        self._load()
-
-    def _load(self):
         try:
-            # ensure version is set
-            self.policy_document['Version'] = '2012-10-17'
+            policy_document = self.load_policy_document()
 
         except self.policy.meta.client.exceptions.NoSuchEntityException:
             # create an empty s3 access policy
-            self.put(policy_document=deepcopy(BASE_S3_ACCESS_POLICY))
+            policy_document = deepcopy(BASE_S3_ACCESS_POLICY)
+            self.put(policy_document=policy_document)
+
+        # ensure version is set
+        policy_document['Version'] = '2012-10-17'
 
         # ensure statements are correctly configured and build a lookup table
-        for stmt in self.policy_document.setdefault('Statement', []):
+        for stmt in policy_document.setdefault('Statement', []):
```

Comment:
You still need the `setdefault()`?

Reply:
It's better than `self.policy_document['Statement'] = self.policy_document.get('Statement', [])` - but it should be on a separate line.